Issue/1528: Define an abstract list table base class, used to extend all list tables. #1528

ramiabraham opened the pull request:
- Defines an abstract list table base class, used to extend all list tables.
- Require base class in main plugin loader.
- See: `admin/class-affwp-list-table.php`

#1528

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+	 * @return void
+	 */
+	public function search_box( $text, $input_id ) {
+		if ( empty( $_REQUEST['s'] ) && !$this->has_items() )
```
Let's please go ahead and add brackets around this.

ramiabraham replied:
(Updated)

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+
+		$input_id = $input_id . '-search-input';
+
+		if ( ! empty( $_REQUEST['orderby'] ) )
```
Also brackets here please

ramiabraham replied:
(Updated)

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+	 * @since 1.9
+	 * @return array $views All the views available for this class.
+	 */
+	public function get_views() {
```
Could we expand this to make registering views as simple as setting up a single array? That way child classes do not need to perform logic to set the current view.

ramiabraham replied:
Added in https://github.com/AffiliateWP/AffiliateWP/pull/1533/commits/469fca21c2ae32b9782178d33ca69a4c03ab669c

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+	 * @since  1.9
+	 * @return array $columns Array of all the list table columns.
+	 */
+	public function get_columns() {
```
Let's break this into two methods, one that is private (or at least marked as such) and contains the `'affwp_list_table_' . $this->singular . '_table_columns'` filter and then one that is public that does nothing more than return the array of column names / keys.

ramiabraham replied:
Added in https://github.com/AffiliateWP/AffiliateWP/pull/1533/commits/c2642f192ff3ec2cf6d050f8137d3ee5411e9a1f

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+	 * @since  1.9
+	 * @return array Array of all the sortable columns.
+	 */
+	public function get_sortable_columns() {
```
Same thing has as mentioned above for get_columns()

ramiabraham replied:
Added in https://github.com/AffiliateWP/AffiliateWP/pull/1533/commits/c2642f192ff3ec2cf6d050f8137d3ee5411e9a1f

pippinsplugins commented on includes/admin/class-affwp-list-table.php:
```diff
@@ -0,0 +1,315 @@
+	 * @since  1.9
+	 * @return array $actions Array of the bulk actions available for this list table.
+	 */
+	public function get_bulk_actions() {
```
Same thing here as mentioned above for get_columns()

ramiabraham replied:
Added in https://github.com/AffiliateWP/AffiliateWP/pull/1533/commits/c2642f192ff3ec2cf6d050f8137d3ee5411e9a1f

ramiabraham commented:
As of https://github.com/AffiliateWP/AffiliateWP/pull/1533/commits/5a227c1a2628fcd23ab519642c6564a5451cffab:

- Above adjustments added
- Simplified `get_views()` method
- Introduced private methods as needed.
